### ISSUE TITLE
It is possible to change a const object

### ISSUE THREAD

**Issue by AlekseyMuraviev** [jbeder/yaml-cpp#633]
yaml-cpp-0.6.2
Microsoft Visual Studio Professional 2015 Version 14.0.25431.01 Update 3

This code will compile
```
const YAML::Node node = YAML::Load("{1: 2}");
cout << node << endl;
YAML::Node tmp;
tmp = node[1];
tmp = 3;
cout << node << endl;
```
and display the following
```
{1: 2}
{1: 3}
```

**Comment by AlekseyMuraviev** [comment]
Yes it is. But, if you try to assign a value directly, then nothing happens:

```
const YAML::Node node = YAML::Load("{1: 2}");
node[1] = 3;
```

> Error	C2678	binary '=': no operator found which takes a left-hand operand of type 'const YAML::Node' (or there is no acceptable conversion)

However, the code below compiles and changes the value:
```
const YAML::Node node = YAML::Load("{1: 2}");
YAML::Node{node[1]} = 3;
```
It looks not very intuitive.

**Comment by TheFox** [comment]
Because `YAML::Node tmp;` in your first example isn't const. In case of `node[1] = 3;` -- `node` is `const`.
